### PR TITLE
fix #17779 - PHP 8 numeric operation

### LINKED ISSUE #17779: "Show All" shows error message
### Describe the bug

When I click the "show all" checkbox, an error message is shown

### To Reproduce

1. Open any table
2. On the Browse tab, tick "Show All"
4. Select "OK" in the confirm dialog ("Do you really want to see all of the rows? For a big table this could crash the browser.")
5. Error is shown

```js
Browse mode Browse mode Customize browse mode. Browse mode Table navigation bar Use only icons, only text or both. Icons Text Both How to display various action links Use only icons, only text or both. Icons Text Both Allow to display all the rows Whether a user should be displayed a "show all (rows)" button. Maximum number of rows to display Number of rows displayed when browsing a result set. If the result set contains more rows, "Previous" and "Next" links will be shown. 25 50 100 250 500 Default sorting order SMART - i.e. descending order for columns of type TIME, DATE, DATETIME and TIMESTAMP, ascending order otherwise. ASC DESC SMART Highlight pointer Highlight row pointed by the mouse cursor. Row marker Highlight selected rows. Grid editing: trigger action Click Double click Disabled Grid editing: save all edited cells at once Repeat headers Repeat the headers every X cells, 0 deactivates this feature. Limit column characters Maximum number of characters shown in any non-numeric column on browse view. Where to show the table row links These are Edit, Copy and Delete links. Nowhere Left Right Both Show row links anyway Whether to show row links even in the absence of a unique key. Primary key default sort order Default sort order for tables with a primary key. None Ascending Descending Remember table's sorting When browsing tables, the sorting of each table is remembered. Relational display For display Options key display column 


if (typeof window.configInlineParams === 'undefined' || !Array.isArray(window.configInlineParams)) {
window.configInlineParams = [];
}
window.configInlineParams.push(function () {
window.Config.registerFieldValidator('MaxRows', 'validatePositiveNumber', true);
window.Config.registerFieldValidator('RepeatCells', 'validateNonNegativeNumber', true);
window.Config.registerFieldValidator('LimitChars', 'validatePositiveNumber', true);

$.extend(window.Messages, {
'error_nan_p': 'Not a positive number!',
'error_nan_nneg': 'Not a non-negative number!',
'error_incorrect_port': 'Not a valid port number!',
'error_invalid_value': 'Incorrect value!',
'error_value_lte': 'Value must be less than or equal to %!',
});

$.extend(window.defaultValues, {
'TableNavigationLinksMode': ['icons'],
'ActionLinksMode': ['both'],
'ShowAll': false,
'MaxRows': ['25'],
'Order': ['SMART'],
'BrowsePointerEnable': true,
'BrowseMarkerEnable': true,
'GridEditing': ['double-click'],
'SaveCellsAtOnce': false,
'RepeatCells': '100',
'LimitChars': '50',
'RowActionLinks': ['left'],
'RowActionLinksWithoutUnique': false,
'TablePrimaryKeyOrder': ['NONE'],
'RememberSorting': true,
'RelationalDisplay': ['K']
});
});
if (typeof window.configScriptLoaded !== 'undefined' && window.configInlineParams) {
window.Config.loadInlineConfig();
Browse mode Browse mode Customize browse mode. Browse mode Table navigation bar Use only icons, only text or both. Icons Text Both How to display various action links Use only icons, only text or both. Icons Text Both Allow to display all the rows Whether a user should be displayed a "show all (rows)" button. Maximum number of rows to display Number of rows displayed when browsing a result set. If the result set contains more rows, "Previous" and "Next" links will be shown. 25 50 100 250 500 Default sorting order SMART - i.e. descending order for columns of type TIME, DATE, DATETIME and TIMESTAMP, ascending order otherwise. ASC DESC SMART Highlight pointer Highlight row pointed by the mouse cursor. Row marker Highlight selected rows. Grid editing: trigger action Click Double click Disabled Grid editing: save all edited cells at once Repeat headers Repeat the headers every X cells, 0 deactivates this feature. Limit column characters Maximum number of characters shown in any non-numeric column on browse view. Where to show the table row links These are Edit, Copy and Delete links. Nowhere Left Right Both Show row links anyway Whether to show row links even in the absence of a unique key. Primary key default sort order Default sort order for tables with a primary key. None Ascending Descending Remember table's sorting When browsing tables, the sorting of each table is remembered. Relational display For display Options key display column 


if (typeof window.configInlineParams === 'undefined' || !Array.isArray(window.configInlineParams)) {
window.configInlineParams = [];
}
window.configInlineParams.push(function () {
window.Config.registerFieldValidator('MaxRows', 'validatePositiveNumber', true);
window.Config.registerFieldValidator('RepeatCells', 'validateNonNegativeNumber', true);
window.Config.registerFieldValidator('LimitChars', 'validatePositiveNumber', true);

$.extend(window.Messages, {
'error_nan_p': 'Not a positive number!',
'error_nan_nneg': 'Not a non-negative number!',
'error_incorrect_port': 'Not a valid port number!',
'error_invalid_value': 'Incorrect value!',
'error_value_lte': 'Value must be less than or equal to %!',
});

$.extend(window.defaultValues, {
'TableNavigationLinksMode': ['icons'],
'ActionLinksMode': ['both'],
'ShowAll': false,
'MaxRows': ['25'],
'Order': ['SMART'],
'BrowsePointerEnable': true,
'BrowseMarkerEnable': true,
'GridEditing': ['double-click'],
'SaveCellsAtOnce': false,
'RepeatCells': '100',
'LimitChars': '50',
'RowActionLinks': ['left'],
'RowActionLinksWithoutUnique': false,
'TablePrimaryKeyOrder': ['NONE'],
'RememberSorting': true,
'RelationalDisplay': ['K']
});
});
if (typeof window.configScriptLoaded !== 'undefined' && window.configInlineParams) {
window.Config.loadInlineConfig();
}

```
### Expected behavior

No error is displayed and all rows are displayed

### Server configuration

- Web server: Apache
- Database version:  10.3.36-MariaDB
- PHP version:  8.1.11
- phpMyAdmin version: 5.3.0-dev+20221010.6785c97d22

### Client configuration

- Browser: Chrome, FF
- Operating system: Windows 11

### PR DESCRIPTION
**Pull request by aliber4079**
php8 throws exception in spite of @ symbol so work around it

### Description

php8 throws exception in spite of @ symbol so work around it

Fixes #17779 

Signed-off-by: Alex Liberman <alex@zoosmart.us>

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: libraries/classes/Display/Results.php [hunk 1]
```diff
@@ -738,9 +738,13 @@ private function getTableNavigation(
...
         $hasRealEndInput = $isInnodb && $this->properties['unlim_num_rows'] > $GLOBALS['cfg']['MaxExactCount'];
-        $posLast = @((int) ceil(
-            (int) $this->properties['unlim_num_rows'] / $_SESSION['tmpval']['max_rows']
-        ) - 1) * intval($_SESSION['tmpval']['max_rows']);
+        if (is_numeric($_SESSION['tmpval']['max_rows'])) {
+            $posLast = @((int) ceil(
+                (int) $this->properties['unlim_num_rows'] / $_SESSION['tmpval']['max_rows']
+            ) - 1) * intval($_SESSION['tmpval']['max_rows']);
+        } else {
+            $posLast = 0;
```
Comment (williamdes) [inline libraries/classes/Display/Results.php:746]: The else statement is ulsess, move the assignment above the if :)

Comment (aliber4079) [reply libraries/classes/Display/Results.php]: Added 😀

**Review by williamdes — APPROVED** [review]
Looks good

**Comment by aliber4079** [comment]
oops sorry I don't know how I did this but can you reverse it

image

**Review by williamdes — APPROVED** [review]

**Review by MauricioFauth — APPROVED** [review]

**Comment by MauricioFauth** [comment]
Merged, thanks for your contribution!

**Pull request merged**